framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -654,6 +656,61 @@ protected JavaExpression visitLocalVariable(
     convertAnnotatedTypeMirror(stringToJavaExpr, atm);
   }
 
+  /**
+   * Delocalizes dependent type annotations in {@code atm} so that they can be placed on the
+   * declaration of the given method or constructor being invoked. Used by whole program inference
+   * to infer dependent types for method/constructor parameters based on the actual arguments used
+   * at call sites.
+   *
+   * @param atm the annotated type mirror to delocalize
+   * @param invocationTree the method or constructor invocation
+   * @param arguments the actual arguments to the method or constructor
+   * @param methodElt the declaration of the method or constructor being invoked
+   */
+  public void delocalizeAtCallsite(
+      AnnotatedTypeMirror atm,
+      Tree invocationTree,
+      List<Node> arguments,
+      ExecutableElement methodElt) {
+
+    if (!hasDependentType(atm)) {
+      return;
+    }
+
+    // For use in stringToJavaExpr below, to avoid re-computation. Especially
+    // important for the TreePath, which is expensive to compute.
+    List<JavaExpression> argsAsExprs =
+        arguments.stream().map(LocalVariable::fromNode).collect(Collectors.toList());
```

Comment:
Use org.plumelib.util.CollectionsPlume#mapList(Function<? super FROM,? extends TO>, Iterable<FROM>)